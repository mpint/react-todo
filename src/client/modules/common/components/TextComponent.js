import React from 'react';
import { isUndefined } from 'lodash';

const TextComponent = (props) => {
  const content = isUndefined(props.content) ? 'a component for testing' : props.content;

  return (
    <div style={ props.style }>
      { content }
    </div>
  );
};

export default TextComponent;
