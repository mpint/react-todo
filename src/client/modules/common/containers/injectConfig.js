import React, { PropTypes } from 'react';
import config from '~/config/main.config';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import muiThemeFile from '~/styles/muiTheme';

const muiTheme = getMuiTheme(muiThemeFile);

function injectConfiguration(BaseComponent) {
  class ConfiguredComponent extends React.Component {
    static childContextTypes = {
      config: PropTypes.object.isRequired,
      muiTheme: PropTypes.object.isRequired
    };

    getChildContext() {
      return { config, muiTheme };
    }

    render() {
      return ( <BaseComponent { ...this.props }/> );
    }
  }

  return ConfiguredComponent;
}

export default injectConfiguration;
