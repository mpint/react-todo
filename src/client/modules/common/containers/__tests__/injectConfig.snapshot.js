import React from 'react';
import renderer from 'react-test-renderer';

import { ConfiguredProvider } from '~/lib/test.helpers';

import injectConfig from '../injectConfig';
import TextComponent from '../../components/TextComponent';

describe('injectConfig', function() {

  it('should render', function() {
    const Thing = injectConfig(TextComponent);

    const component = renderer.create(
      <ConfiguredProvider>
        <Thing/>
      </ConfiguredProvider>
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
