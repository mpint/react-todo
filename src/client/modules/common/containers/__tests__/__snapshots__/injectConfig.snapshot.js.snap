exports[`injectConfig should render 1`] = `
<div
  style={undefined}>
  a component for testing
</div>
`;
